fury/tests/test_material.py

```diff
@@ -428,428 +169,196 @@ def test_manifest_pbr(interactive=False):
+        ss = load_image(tmp_fname)
+        actual = ss[75, 100, :] / 1000
+        desired = np.array([170, 170, 255]) / 1000
+        npt.assert_array_almost_equal(actual, desired, decimal=2)
+        actual = ss[125, 125, :] / 1000
+        npt.assert_array_almost_equal(actual, desired, decimal=2)
+        actual = ss[125, 75, :] / 1000
+        desired = np.array([85, 85, 170]) / 1000
+        npt.assert_array_almost_equal(actual, desired, decimal=2)
+
+        # Test specular power
+        material.manifest_standard(test_actor, specular_level=1,
+                                   specular_power=5)
+        window.record(scene, out_path=tmp_fname, size=(200, 200),
+                      reset_camera=True)
+        npt.assert_equal(os.path.exists(tmp_fname), True)
+        ss = load_image(tmp_fname)
+        actual = ss[75, 100, :] / 1000
+        desired = np.array([34, 34, 204]) / 1000
+        npt.assert_array_almost_equal(actual, desired, decimal=2)
+        actual = ss[125, 125, :] / 1000
+        npt.assert_array_almost_equal(actual, desired, decimal=2)
+        actual = ss[125, 75, :] / 1000
+        desired = np.array([1, 1, 86]) / 1000
+        npt.assert_array_almost_equal(actual, desired, decimal=2)
+
+        # Test specular color
+        material.manifest_standard(test_actor, specular_level=1,
+                                   specular_color=(1, 0, 0), specular_power=5)
+        window.record(scene, out_path=tmp_fname, size=(200, 200),
+                      reset_camera=True)
+        npt.assert_equal(os.path.exists(tmp_fname), True)
+        ss = load_image(tmp_fname)
+        actual = ss[75, 100, :] / 1000
+        desired = np.array([34, 0, 170]) / 1000
+        npt.assert_array_almost_equal(actual, desired, decimal=2)
+        actual = ss[125, 125, :] / 1000
+        npt.assert_array_almost_equal(actual, desired, decimal=2)
+        actual = ss[125, 75, :] / 1000
+        desired = np.array([1, 0, 85]) / 1000
+        npt.assert_array_almost_equal(actual, desired, decimal=2)
+
+        scene.clear()  # Reset scene
+
+        # Special case: Contour from roi
+        data = np.zeros((50, 50, 50))
+        data[20:30, 25, 25] = 1.
+        data[25, 20:30, 25] = 1.
+        test_actor = actor.contour_from_roi(data, color=np.array([1, 0, 1]))
+        scene.add(test_actor)
+
+        window.record(scene, out_path=tmp_fname, size=(200, 200),
+                      reset_camera=True)
+        npt.assert_equal(os.path.exists(tmp_fname), True)
+        ss = load_image(tmp_fname)
+        actual = ss[90, 110, :] / 1000
+        desired = np.array([253, 0, 253]) / 1000
+        npt.assert_array_almost_equal(actual, desired, decimal=2)
+        actual = ss[90, 60, :] / 1000
+        desired = np.array([180, 0, 180]) / 1000
+        npt.assert_array_almost_equal(actual, desired, decimal=2)
+
+        material.manifest_standard(test_actor)
+        window.record(scene, out_path=tmp_fname, size=(200, 200),
```

Comment:
Interesting that you prefer using record than snapshot. Snapshot returns directly the framebuffer as a numpy array.

Reply:
Thanks, I didn't think of it. However, I found this problem (#467) when using `window.snapshot()`.